framework/src/base/FEProblemBase.C

```diff
@@ -364,6 +364,8 @@ void FEProblemBase::initialSetup()
   if (_app.isRecovering() && (_app.isUltimateMaster() || _force_restart))
     _resurrector->setRestartFile(_app.getRecoverFileBase());
 
+  _mesh.initialSetup();
```

Comment:
Any forethought to your location choice in this method? It's just that we spend a lot of time moving logic around in this method. I wonder if this is the right place to handle all the recover/restart cases that we will surely encounter 😉